src/calendar.cpp

```diff
@@ -556,11 +556,24 @@ time_duration rng( time_duration lo, time_duration hi )
 
 season_type season_of_year( const time_point &p )
 {
-    if( calendar::eternal_season() ) {
-        // If we use calendar::start to determine the initial season, and the user shortens the season length
-        // mid-game, the result could be the wrong season!
-        return calendar::initial_season;
+    static int prev_turn = -1;
```

Comment:
Why are you storing the turn number and not a `time_point` instance?

Reply:
I think I wrote this up after a 14 hour shift of physical labor.